test_mp_allreduce/test_paddle_incubate.py

```diff
@@ -0,0 +1,267 @@
+        del out_eager_baseline
+        del out_grads_eager_baseline
+        paddle.device.cuda.empty_cache()
+
+        for i in range(50):
+            out_eager, out_grads_eager = self._cal_eager_res(x_eager, dout_eager)
+            out_eager = out_eager.numpy()
+            out_grads_eager = out_grads_eager[0].numpy()
+            if paddle.distributed.get_rank() == 0:
+                np.testing.assert_equal(
+                    out_eager,
+                    out_eager_baseline_np,
+                    err_msg=(
+                        'Develop: mp_allreduce eager forward is unstable in %s dtype'
+                    )
+                    % self._dtype,
+                )
+                np.testing.assert_equal(
+                    out_grads_eager,
+                    out_grads_eager_baseline_np,
+                    err_msg=(
+                        'Develop: mp_allreduce eager grad is unstable in %s dtype'
+                    )
+                    % self._dtype,
+                )
+
+    def _test_static_stability(self):
+        with paddle.fluid.framework._dygraph_guard(None):
+            mp, sp = paddle.static.Program(), paddle.static.Program()
+            with paddle.static.program_guard(mp, sp):
+                x_static, dout_static = self._gen_static_inputs_and_dout()
+                (out_static_pg, out_grads_static_pg) = self._cal_static_res(
+                    x_static,
+                    dout_static,
+                )
+            exe = paddle.static.Executor()
+            exe.run(sp)
+            out = exe.run(
+                mp,
+                feed={"x": self._np_x, "dout": self._np_dout},
+                fetch_list=[out_static_pg] + out_grads_static_pg,
+            )
+            out_static_baseline, out_grads_static_baseline = out[0], out[1:]
+            for i in range(50):
+                out = exe.run(
+                    mp,
+                    feed={"x": self._np_x, "dout": self._np_dout},
+                    fetch_list=[out_static_pg] + out_grads_static_pg,
+                )
+                out_static, out_grads_static = out[0], out[1:]
+                if paddle.distributed.get_rank() == 0:
+                    np.testing.assert_equal(
+                        out_static,
+                        out_static_baseline,
+                        err_msg=(
+                            'Develop: mp_allreduce static forward is unstable in %s dtype'
+                        )
+                        % self._dtype,
+                    )
+                    np.testing.assert_equal(
+                        out_grads_static[0],
+                        out_grads_static_baseline[0],
+                        err_msg=(
+                            'Develop: mp_allreduce static grad is unstable in %s dtype'
+                        )
+                        % self._dtype,
+                    )
+
+shape_list =[[1, 12288], [1, 4096, 12288]]
+dtype_list = ["float32", "float16"]
```

Comment:
do not support bfp16?

Reply:
_mp_allreduce doesn't support  bfp16, When use_calc_stream is True.